iroha_p2p/src/peer.rs

```diff
@@ -0,0 +1,413 @@
+use async_stream::stream;
+use futures::Stream;
+use iroha_actor::{Actor, Addr, Context, Handler};
+use iroha_logger::warn;
+use parity_scale_codec::{Decode, Encode};
+use rand::{Rng, RngCore};
+use tokio::{
+    io,
+    io::{AsyncReadExt, AsyncWriteExt},
+    net::{
+        tcp::{OwnedReadHalf, OwnedWriteHalf},
+        TcpStream,
+    },
+};
+use ursa::{
+    encryption::symm::{Encryptor, SymmetricEncryptor},
+    kex::KeyExchangeScheme,
+    keys::{PrivateKey, PublicKey},
+};
+
+use crate::{
+    message::{Message, MessageResult},
+    network::{Post, Received},
+    Error, Network,
+};
+
+const MAX_MESSAGE_LENGTH: usize = 2 * 1024 * 1024;
+const MAX_HANDSHAKE_LENGTH: usize = 255;
+pub const DEFAULT_AAD: &[u8; 12] = b"Iroha2Iroha2";
```

Comment:
Please add a comment for this abbreviation. Is it `Additional Authorization Details`?